### ISSUE TITLE
Launcher icons disappear (issue reocurrence with v7)

### ISSUE THREAD

**Issue by smithfred** [jderose9/dash-to-panel#92]
Within about 10 minutes of upgrading to v7 I've already seen the application launcher icons disappear once (only disabling then re-enabling the extension brought them back). I saw this previously with a pre-v6 version, not sure if there was an issue ticket for that?

My panel size is 32 if that helps; I seem to remember that was relevant last time.

**Comment by jderose9** [comment]
This is the first report of launcher icons disappearing. You mean the dash items (favorites/running applications) rather than the notification icons or status menu, correct?

Which distro and gnome version are you using? If you can grab a screenshot that would be great. 

Which other extensions are you using? Can you disable all other extensions to verify it is an issue with this one?

One other possibly useful piece of info would be the log.. After you experience this open the terminal and type:
`journalctl /usr/bin/gnome-shell -o cat --since "5 minutes ago"`

**Comment by smithfred** [comment]
It happened for the second time just now, before I read your comment. But it just looks normal except with no launcher icons (yes, favourites/running applications) - the same as if they were all removed/closed.

F25/GNOME 3.22, Impatience, Panel osd, Windowoverlay icons, Workspaces to dock. Not much point in disabling them since the bug is infrequent enough that it would take weeks to determine whether any were causing a problem and then bisect to a theoretical culprit, by which time I would have gone insane using otherwise-stock GNOME Shell ;)

journalctl gave lots of these:

st_widget_get_theme_node called on the widget [0x5639a31f5c90 StIcon:last-child first-child] which is not in the stage.
 (with differing widget IDs)

Then loads of these a few minutes later, which I *think* corresponds to the time the icons disappeared (by checking timestamps in journalctl without -o cat):

st_widget_get_theme_node called on the widget [0x5639a652c9e0 Gjs_DashItemContainer.dash-item-container:last-child ("-1")] which is not in the stage.
st_widget_get_theme_node called on the widget [0x5639a650e500 StButton.app-well-app running1:last-child first-child ("-1")] which is not in the stage.
st_widget_get_theme_node called on the widget [0x5639a650fda0 StWidget:last-child first-child ("-1")] which is not in the stage.
st_widget_get_theme_node called on the widget [0x5639a6510950 StBin.overview-icon] which is not in the stage.
st_widget_get_theme_node called on the widget [0x5639a65113d0 ShellGenericContainer:last-child first-child] which is not in the stage.

**Comment by jderose9** [comment]
Those pretty much just mean the icon isn't visible, which you already know :) If something was in the log that was helpful it would probably be right before that.

Do you know what you are doing directly before the issue occurs? For instances, closing the activities overview, or some other specific action?

**Comment by smithfred** [comment]
Nope to both questions unfortunately. If some common factor emerges I'll add it here.

**Comment by smithfred** [comment]
Has occurred a couple of times since, no obvious pattern still.

**Comment by serenase** [comment]
Antergos, Gnome 3.22.2, Dash to panel 8.1 (from AUR).
I lost icons from panel everytime I open dash and get them back by restarting the blugin or lock out and back in. This happens both Wayland and Xorg sessions.

`Allocating size to ShellEmbeddedWindow 0x4a4a5d0 without calling gtk_widget_get_
_cogl_buffer_fini: assertion '!(buffer->flags & COGL_BUFFER_FLAG_MAPPED)' failed
gsignal.c:2641: instance '0x2fcdc30' has no handler with id '9493'
clutter_layout_manager_get_child_meta: assertion 'CLUTTER_IS_LAYOUT_MANAGER (man
g_object_set: assertion 'G_IS_OBJECT (object)' failed
g_object_set: assertion 'G_IS_OBJECT (object)' failed
clutter_layout_manager_get_child_meta: assertion 'CLUTTER_IS_LAYOUT_MANAGER (man
g_object_set: assertion 'G_IS_OBJECT (object)' failed
g_object_set: assertion 'G_IS_OBJECT (object)' failed
clutter_layout_manager_get_child_meta: assertion 'CLUTTER_IS_LAYOUT_MANAGER (man
g_object_set: assertion 'G_IS_OBJECT (object)' failed
g_object_set: assertion 'G_IS_OBJECT (object)' failed
clutter_layout_manager_get_child_meta: assertion 'CLUTTER_IS_LAYOUT_MANAGER (man
g_object_set: assertion 'G_IS_OBJECT (object)' failed
g_object_set: assertion 'G_IS_OBJECT (object)' failed
clutter_layout_manager_get_child_meta: assertion 'CLUTTER_IS_LAYOUT_MANAGER (man
g_object_set: assertion 'G_IS_OBJECT (object)' failed
g_object_set: assertion 'G_IS_OBJECT (object)' failed
clutter_layout_manager_get_child_meta: assertion 'CLUTTER_IS_LAYOUT_MANAGER (man
g_object_set: assertion 'G_IS_OBJECT (object)' failed
g_object_set: assertion 'G_IS_OBJECT (object)' failed
clutter_layout_manager_get_child_meta: assertion 'CLUTTER_IS_LAYOUT_MANAGER (man
g_object_set: assertion 'G_IS_OBJECT (object)' failed
lines 1-23
`

**Comment by jderose9** [comment]
Hmm yeah I'm not seeing this on Fedora but I just installed Antergos and am seeing it quite frequently to the point that it's not really usable.

**Comment by joshgh** [comment]
I am using Ubuntu Gnome 17.04 and the dash icons disappear seemingly randomly, I can sometimes trigger the disappearance by opening the activities overview, the icons reappear when I hover over the spot in the panel where they should be.

```bash
JS WARNING: [resource:///org/gnome/shell/ui/messageList.js 392]: reference to undefined property this._useBodyMarkup
JS WARNING: [resource:///org/gnome/shell/ui/messageList.js 169]: reference to undefined property this._container
Invalid result meta returned from search provider org.gnome.Software.desktop
Invalid result meta returned from search provider org.gnome.Software.desktop
_cogl_buffer_fini: assertion '!(buffer->flags & COGL_BUFFER_FLAG_MAPPED)' failed
Received error from DBus search provider org.gnome.Contacts.desktop: Gio.DBusError: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name :1.713 was not provided by any .service fil
_cogl_buffer_fini: assertion '!(buffer->flags & COGL_BUFFER_FLAG_MAPPED)' failed
_cogl_buffer_fini: assertion '!(buffer->flags & COGL_BUFFER_FLAG_MAPPED)' failed
Invalid result meta returned from search provider org.gnome.Software.desktop
_cogl_buffer_fini: assertion '!(buffer->flags & COGL_BUFFER_FLAG_MAPPED)' failed
Invalid result meta returned from search provider org.gnome.Software.desktop
_cogl_buffer_fini: assertion '!(buffer->flags & COGL_BUFFER_FLAG_MAPPED)' failed
_cogl_buffer_fini: assertion '!(buffer->flags & COGL_BUFFER_FLAG_MAPPED)' failed
Received error from DBus search provider org.gnome.Calculator.desktop: Gio.DBusError: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name :1.779 was not provided by any .service f
Invalid result meta returned from search provider org.gnome.Software.desktop
_cogl_buffer_fini: assertion '!(buffer->flags & COGL_BUFFER_FLAG_MAPPED)' failed
Received error from DBus search provider org.gnome.Photos.desktop: Gio.DBusError: GDBus.Error:org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without reply
_cogl_buffer_fini: assertion '!(buffer->flags & COGL_BUFFER_FLAG_MAPPED)' failed
```

**Comment by ckrzen** [comment]
Hrmmmmm ... I just observed this on "Ubuntu GNOME 16.04.2" when returning from the lock-screen. Admittedly, the system is low on RAM(only 4GB), but it was the first time I've seen it and there were not a lot of other apps open. However -- I would like to mention that Evolution seems to consume a lot of RAM with IMAPx accounts and it was open, minimized.

**Comment by patrickpcs91** [comment]
The same here on both v8 and v9. Always when we came back from the lock-screen. I'm deploying it in maybe 30 workstation and it happens independently about the amount of RAM. From 2 to 8 GB of RAM all the same behavior. All the stations running Ubuntu Gnome 16.04.3 LTS. BTW, where can I get those logs for you?

**Comment by mrperson2015** [comment]
I get this every time I (lock/unlock) login to Ubuntu.

Memory: 64 Gib (38.7 free)
Processor: Intel® Core™ i7-7700K CPU @ 4.20GHz × 8 
Graphics: Intel® Kabylake GT2 
Base system: Ubuntu 16.04.3 LTS 64-bit
Disk: 424.8 GB
Gnome Version: GNOME Shell 3.18.5

**Comment by daledavies** [comment]
This happens for me occasionally too, not every time but usually when coming back from the lock screen. 

Ubuntu 17.10
Nvidia v3849 driver

**Comment by smithfred** [comment]
This is worse on v11/GNOME 3.26 - now even reloading (or disabling/re-enabling) the extension doesn't restore the icons; when they disappear they're gone until Shell is reloaded.

**Comment by smithfred** [comment]
When this happens, if I disable DTP, the standard GS app bar doesn't show non-pinned, running items (only pinned items whether running or not) so there seems to be a related problem with GS.

**Comment by smithfred** [comment]
Upstream bug filed for new 3.26 GS bugginess seen after disabling DtP: https://bugzilla.gnome.org/show_bug.cgi?id=790766

Not sure if the bug in Shell's Dash is caused by DtP (maybe its state is corrupted by DtP and persists even after disabling the extension?)

**Comment by jderose9** [comment]
Closing this as I believe it has been resolved. Please post back if you find otherwise in the latest master (or after v12 is released). Thanks!

**Comment by zholmes1** [comment]
This has been happening every time I unlock. Ubuntu 18.04. Opening Dash to Panel settings seems to fix it 75% of the time.

**Comment by skeletorkun** [comment]
still happening

**Comment by rgayler** [comment]
This is also happening for me. At some random time in a session I notice that the launcher icons in the dock have disappeared. The space where they used to be is just a vacant strip of desktop. If I maximise a window it occupies the space formerly occupied by the launcher icons. At some random time later in the session the launcher items may have reappeared. I haven't been able to spot any correlation with what I was doing before the icons disappeared.

I have only noticed this happening in the last couple of weeks.

I am running Ubuntu18.04 64-bit and am not aware that I have made any Gnome-related changes. I only see this problem on my desktop PC with a 4K monitor. I don't see it happening on my Ubuntu 18.04 laptop.
